src/openrct2/actions/RideSetSetting.hpp

```diff
@@ -0,0 +1,331 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../ride/Ride.h"
+#include "../ride/RideData.h"
+#include "GameAction.h"
+
+enum class RideSetSetting : uint8_t
+{
+    Mode,
+    Departure,
+    MinWaitingTime,
+    MaxWaitingTime,
+    Operation,
+    InspectionInterval,
+    Music,
+    MusicType,
+    LiftHillSpeed,
+    NumCircuits,
+    RideType,
+};
+
+DEFINE_GAME_ACTION(RideSetSettingAction, GAME_COMMAND_SET_RIDE_SETTING, GameActionResult)
+{
+private:
+    NetworkRideId_t _rideIndex{ -1 };
+    uint8_t _setting{ 0 };
+    uint8_t _value{ 0 };
+
+public:
+    RideSetSettingAction()
+    {
+    }
+    RideSetSettingAction(ride_id_t rideIndex, RideSetSetting setting, uint8_t value)
+        : _rideIndex(rideIndex)
+        , _setting(static_cast<uint8_t>(setting))
+        , _value(value)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags();
```

Comment:
GA_FLAGS::ALLOW_WHILE_PAUSED missing

Reply:
I don't think you should be able to change these settings whilst paused.